drivers/Serial.c

```
@@ -37,8 +38,43 @@
+static int SerialDevWrite(struct CharDevice* dev, const void* buffer, uint32_t size) {
+    if (!serial_initialized) return -1;
+    const char* buf = (const char*)buffer;
+    for (uint32_t i = 0; i < size; i++) {
+        if (SerialWriteChar(buf[i]) < 0) {
+            return i;
+        }
+    }
+    return size;
+}
+
+static CharDevice_t g_serial_device = {
```

Clang (14.0.6)

[warning] variable 'g_serial_device' is non-const and globally accessible, consider making it const

(cppcoreguidelines-avoid-non-const-global-variables)